apps/editing-toolkit/editing-toolkit-plugin/coming-soon/coming-soon.php

```diff
@@ -31,6 +35,42 @@ function should_show_coming_soon_page() {
 	return apply_filters( 'a8c_show_coming_soon_page', $should_show );
 }
 
+/**
+ * Determines whether the coming soon page should be bypassed.
+ *
+ * It checks if share code is provided as GET parameter, or as a cookie.
+ * Then it validates the code against blog option, and if sharing code is valid,
+ * it allows bypassing the Coming Soon page.
+ *
+ * Finally, it sets a code in cookie and sets header that prevents robots from indexing.
+ *
+ * @return bool
+ */
+function is_accessed_by_valid_share_link() {
```

Comment:
Can we add some tests for this method?

Reply:
Good point. I've added tests for this method and for the new one that I extracted from the existing one.